Cortex/cortex_module/cortex_edr_connector.py

```diff
@@ -0,0 +1,170 @@
+                "filters": [{"field": "creation_time", "operator": "gte", "value": 0}],
+                "search_from": 0,
+                "search_to": 0,
+                "sort": {"field": "creation_time", "keyword": "desc"},
+            }
+        }
+        self._timestamp_cursor = ""
+
+    @property
+    def timestamp_cursor(self) -> int:
+        now = datetime.now(timezone.utc)
+
+        with self.context as cache:
+            timestamp_cursor_str = cache.get("timestamp_cursor")
+
+            if timestamp_cursor_str is None:
+                before_two_days = now - timedelta(days=2)
+                return int(before_two_days.timestamp())
+
+            timestamp_cursor = int(isoparse(timestamp_cursor_str).timestamp())
+
+            return timestamp_cursor
+
+    @timestamp_cursor.setter
+    def timestamp_cursor(self, time: int) -> None:
+        if len(str(time)) == 13:
+            time_to_iso8601 = datetime.utcfromtimestamp(time / 1000.0).isoformat()
+        else:
+            time_to_iso8601 = datetime.utcfromtimestamp(time).isoformat()
+        self._timestamp_cursor = time_to_iso8601
+        with self.context as cache:
+            cache["timestamp_cursor"] = self._timestamp_cursor
+
+    @cached_property
+    def alert_url(self) -> str:
+        return f"https://api-{self.module.configuration.fqdn}/public_api/v1/alerts/get_alerts_multi_events"
+
+    @cached_property
+    def pagination_limit(self) -> int:
+        return max(self.configuration.chunk_size, 100)
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            self.module.configuration.api_key,
+            self.module.configuration.api_key_id,
+        )
+
+    def split_alerts_events(self, alerts: List[Any]) -> List[str]:
+        """Split events from alerts and put them in the same list"""
+
+        combined_data = []
+        for alert in alerts:
+            shared_id = alert.get("alert_id")
+            events = alert.get("events")
+            del alert["events"]
+            combined_data.append(orjson.dumps(alert).decode("utf-8"))
+            for event in events:
+                event["external_id"] = shared_id
+                combined_data.append(orjson.dumps(event).decode("utf-8"))
+
+        return combined_data
+
+    def get_alerts_events_by_offset(self, offset: int, creation_time: int, pagination: int) -> Tuple[int, List[Any]]:
+        """Requests the Cortex API using the offset"""
+
+        search_from, serch_to = offset, offset + pagination
+        self.query["request_data"]["search_from"] = search_from
+        self.query["request_data"]["search_to"] = serch_to
+        self.query["request_data"]["filters"][0]["value"] = creation_time
+
+        response_query = self.client.post(url=self.alert_url, json=self.query).json().get("reply")
+        combined_data = self.split_alerts_events(response_query.get("alerts"))
+
+        return response_query.get("total_count"), combined_data
```

Comment:
Is `total_count` optional in the response?
If no:
```suggestion
        return response_query["total_count"], combined_data
```
If yes, to avoid having `total_count = None` (that the rest of the code will not accept):
```suggestion
        return response_query.get("total_count", 0), combined_data
```